test/runtests.jl

```diff
@@ -1,136 +1,143 @@
-        @test grid.substations[1].population == 6193
-        @test grid.substations[47].generation == 16.1
-        @test test_connectivity(grid.sub_conn, false)
-        @test total_links(grid.sub_conn) == 69
-        @test mean_shortest_path(sub_connectivity(grid) .> 0) ≈ 4.107741059302851
-        @test mean_shortest_path(
-            (sub_connectivity(grid) .> 0),
-            distance(substations(grid))
-            ) ≈ 113.21283328057127
-        @test robustness_line(grid.sub_conn, 10) > 1
-        @test robustness_node(grid.sub_conn, 10) > 1
-        count = SyntheticGrids.count_bus_type(grid1)
-        cluster!(
-            grid1,
-            round(Int, 0.3 * count[1]),
-            round(Int, 0.05 * count[2]),
-            round(Int, 0.5 * count[2])
-        )
-        @test (grid == grid1)
+            add_gen!(grid, (11., 11.), 100, [11], ["ki"], reconnect = false)
+            add_load!(grid, (19.,12.), 100, 100, 100, reconnect = true)
+            add_substation!(
+                grid,
+                (11., 18.),
+                [11],
+                1000,
+                6000,
+                500,
+                Set(),
+                [grid.buses[139], grid.buses[138]]
+            )
+            @test size(grid.bus_conn) == (139, 139)
+            @test size(grid.sub_conn) == (44, 44)
+        end
+
+        @testset "Clustering grids" begin
+            grid = baseline_grid()
+            count = SyntheticGrids.count_bus_type(grid)
+            cluster!(
+                grid,
+                round(Int, 0.3 * count[1]),
+                round(Int, 0.05 * count[2]),
+                round(Int, 0.5 * count[2])
+            )
+            @test length(substations(grid)) == 47
+            @test grid.substations[1].population == 6193
+            @test grid.substations[47].generation == 16.1
+            @test test_connectivity(grid.sub_conn, false)
+            @test total_links(grid.sub_conn) == 70
+            @test mean_shortest_path(sub_connectivity(grid) .> 0) ≈ 4.1620642824807605
+            @test mean_shortest_path(
+                (sub_connectivity(grid) .> 0),
+                distance(substations(grid))
+                ) ≈ 111.6094877525516
+            @test robustness_line(grid.sub_conn, 10) > 1
+            @test robustness_node(grid.sub_conn, 10) > 1
+        end
+
+        @testset "Test seeding and object comparison" begin
+            grid = baseline_grid()
+            grid1 = baseline_grid()
+
+            @test grid == grid1
+        end
     end
 
     @testset "Input and Output" begin
         @testset "Pandapower interface" begin
+            grid = baseline_grid()
             pgrid = to_pandapower(grid, DUMMYPPC)
-            @test length(pgrid[:trafo]) == 36
+            @test length(pgrid.trafo) == 31
             pgrid = SyntheticGrids.load_pp_grid(DUMMYPPC)
-            @test length(pgrid[:trafo]) == 36
+            @test length(pgrid.trafo) == 31
+            rm(DUMMYPPC)
```

Comment:
If this testset were to fail and this file was not removed what would happen?

Reply:
Nothing really, you'd just have the artifact from the test left over. We just write to this file so that we can subsequently load from it to ensure that PandaPower is saving/opening properly.

If the testset fails, every other test would overwrite this artifact in line 118.